Reflections/R2 - April 23.md

```diff
@@ -0,0 +1,18 @@
+## 1. What have you accomplished this week? (Please list each accomplishment, and explain)
+
+Over the last week, the largest coal that I have accomplished is the implementation of my modular structure building system. The player is now able to select the structure they would like to build from the structures backstage menu, and then place the structure that they have chosen on any non occupied hex tile in the game map. When they attempt to do this, the system does the math to make sure that the x is unoccupied, and that the player has enough materials in their inventory necessary to build the structure once it's built and placed onto the game map it will function in the same way that the example structures from the proof-of-concept worked. I've also built this system in a way that adding in additional structures will be a relatively simple process.
+
+
+## 2. What challenges or difficulties did you face? If you solved them, how? If not, what have you learned so far? Have you sought help or other resources?
+
+One challenge that I discovered in the process of implementing the system, is that creating the pop-up dialog that will allow the player to create different recipes from the same structure is a more involved process than I was hoping it would be. I've done some research into how to implement this system, and I've discovered that in order to do it it is going to be its own major milestone that will take a significant amount of work to complete as such I'm going to create a detailed GitHub issue to reflect this and give it the proper tags.
```

Comment:
Great job on not immediately following the "rabbit hole" on this one. Realizing it's a large task and framing future work in the context of the larger goal is likely the perfect approach for this.